ethereum/jsonrpc/src/test/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/methods/privacy/eea/EeaGetTransactionCountTest.java

```diff
@@ -0,0 +1,118 @@
+    final String privateFrom = "thePrivateFromKey";
+
+    final Map<String, Object> jsonMap = new HashMap<>();
+    jsonMap.put("address", address.toString());
+    jsonMap.put("privateFrom", privateFrom);
+    jsonMap.put("privateFor", privateFor);
+
+    request = new JsonRpcRequest("2.0", "eea_getTransactionCount", new Object[] {jsonMap});
+  }
+
+  @Test
+  public void validRequestProducesExpectedNonce() throws Exception {
+    PrivacyGroup[] returnedGroups =
+        new PrivacyGroup[] {
+          new PrivacyGroup("Group1", Type.LEGACY, "Group1_Name", "Group1_Desc", new String[0]),
+        };
+
+    when(enclave.findPrivacyGroup(any())).thenReturn(returnedGroups);
+    when(privateTransactionHandler.getSenderNonce(address, "Group1")).thenReturn(8L);
+
+    final EeaGetTransactionCount method =
+        new EeaGetTransactionCount(new JsonRpcParameter(), enclave, privateTransactionHandler);
+
+    final JsonRpcResponse response = method.response(request);
+    assertThat(response).isInstanceOf(JsonRpcSuccessResponse.class);
+
+    final JsonRpcSuccessResponse successResponse = (JsonRpcSuccessResponse) response;
+    assertThat(successResponse.getResult()).isEqualTo("0x8");
+  }
+
+  @Test
+  public void moreThanOneMatchingLegacyGroupReturnsError() throws Exception {
+    PrivacyGroup[] returnedGroups =
+        new PrivacyGroup[] {
+          new PrivacyGroup("Group1", Type.LEGACY, "Group1_Name", "Group1_Desc", new String[0]),
+          new PrivacyGroup("Group2", Type.LEGACY, "Group2_Name", "Group2_Desc", new String[0]),
+        };
+
+    when(enclave.findPrivacyGroup(any())).thenReturn(returnedGroups);
+    when(privateTransactionHandler.getSenderNonce(address, "Group1")).thenReturn(8L);
+
+    final EeaGetTransactionCount method =
+        new EeaGetTransactionCount(new JsonRpcParameter(), enclave, privateTransactionHandler);
+
+    final JsonRpcResponse response = method.response(request);
+    assertThat(response).isInstanceOf(JsonRpcErrorResponse.class);
+  }
+
+  @Test
+  public void noMatchingLegacyGroupsReturnsError() throws Exception {
+    PrivacyGroup[] returnedGroups = new PrivacyGroup[] {};
+
+    when(enclave.findPrivacyGroup(any())).thenReturn(returnedGroups);
+
+    final EeaGetTransactionCount method =
+        new EeaGetTransactionCount(new JsonRpcParameter(), enclave, privateTransactionHandler);
+
+    final JsonRpcResponse response = method.response(request);
+    assertThat(response).isInstanceOf(JsonRpcErrorResponse.class);
+  }
+
+  @Test
+  public void enclaveThrowingAnExceptionResultsInErrorResponse() throws Exception {
+    when(enclave.findPrivacyGroup(any())).thenThrow(new Exception("Enclave Failed"));
+    final EeaGetTransactionCount method =
+        new EeaGetTransactionCount(new JsonRpcParameter(), enclave, privateTransactionHandler);
+
+    final JsonRpcResponse response = method.response(request);
+    assertThat(response).isInstanceOf(JsonRpcErrorResponse.class);
```

Comment:
ditto, is any JsonRpcErrorResponse acceptable? or do you expecting a specific message?

Reply:
Done.